internal/storage/storage.go

```diff
@@ -0,0 +1,279 @@
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/lezhnevay/go_final_project_lezhnev/configs"
+	"github.com/lezhnevay/go_final_project_lezhnev/internal/tasks"
+)
+
+type Store struct {
+	db *sql.DB
+}
+
+// Открываем/создаем_БД
+func OpenDataBase() *sql.DB {
+	// Определяем директорию приложения и проверяем наличие базы данных
+	filePath, err := os.Executable()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbFile := filepath.Join(filepath.Dir(filePath), "scheduler.db")
+
+	envFile := os.Getenv("TODO_DBFILE")
+	if len(envFile) > 0 {
+		dbFile = envFile
+	}
+	log.Println("Путь к базе данных", dbFile)
+	_, err = os.Stat(dbFile)
+
+	var install bool
+	if err != nil {
+		install = true
+	}
+	// если install равен true, после открытия БД требуется выполнить
+	// sql-запрос с CREATE TABLE и CREATE INDEX
+	db, err := sql.Open("sqlite", "scheduler.db")
+	if err != nil {
+		log.Fatal(err)
+	}
+	//defer db.Close()
+	// создаем таблицу и индекс
+	if install {
+
+		_, err = db.Exec(`CREATE TABLE IF NOT EXISTS scheduler (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			date CHAR(8) NOT NULL,
+			title TEXT NOT NULL,
+			comment TEXT,
+			repeat VARCHAR(128) NOT NULL
+			);`)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		_, err = db.Exec(`CREATE INDEX IF NOT EXISTS scheduler_date ON scheduler (date);`)
+		if err != nil {
+			log.Fatal(err)
+		}
+		log.Println("База данных создана")
+	} else {
+		log.Println("База данных была создана ранее")
+	}
+	return db
+
+}
+
+func NewStore(db *sql.DB) Store {
+	return Store{db: db}
+}
+
+// Добавление задачи
+func (s *Store) CreateTask(t configs.Task) (string, error) {
+	var err error
+
+	if t.Title == "" {
+		return "", fmt.Errorf(`{"error":"не указан заголовок задачи"}`)
+	}
+
+	// Проверяем наличие даты
+	if t.Date == "" {
+		t.Date = time.Now().Format(configs.DateFormat)
+	}
+
+	_, err = time.Parse(configs.DateFormat, t.Date)
+	if err != nil {
+		return "", fmt.Errorf(`{"error":"некорректный формат даты"}`)
+	}
+	// Если дата меньше сегодняшней, устанавливаем следующую дату по правилу
+	if t.Date < time.Now().Format(configs.DateFormat) {
+		if t.Repeat != "" {
+			nextDate, err := tasks.NextDate(time.Now(), t.Date, t.Repeat)
+			if err != nil {
+				return "", fmt.Errorf(`{"error":"Некорректное правило повторения"}`)
+			}
+			t.Date = nextDate
+		} else {
+			t.Date = time.Now().Format(configs.DateFormat)
+		}
+	}
+
+	// Добавляем задачу в базу
+	query := `INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)`
+	result, err := s.db.Exec(query, t.Date, t.Title, t.Comment, t.Repeat)
+	if err != nil {
+		return "", fmt.Errorf(`{"error":"не удалось добавить задачу"}`)
```

Comment:
Теряешь ошибки, тебе же плохо потом дебажить
или в лог, или так
```suggestion
		return "", fmt.Errorf(`{"error":"не удалось добавить задачу, %v", err}`)
```
Вообще ты тут миксуешь ошибку с формированием ответа клиенту, в идеале ошибку так обернуть
```suggestion
		return "", fmt.Errorf("error while adding tasl: %w", err)
```
А уже в апи формировать ответ

Reply:
Соответственно у тебя это много где так